apps/mobile/src/hooks/useDelegateKey.ts

```diff
@@ -0,0 +1,117 @@
+import { useState, useCallback } from 'react'
+import { ethers } from 'ethers'
+import DeviceCrypto from 'react-native-device-crypto'
+import * as Keychain from 'react-native-keychain'
+
+import Logger from '../utils/logger'
+import { Address } from '@/src/types/address'
+import GatewayService from '../services/gateway/GatewayService'
+import { useSign } from './useSign'
+import { asymmetricKey, keychainGenericPassword } from '@/src/store/constants'
+
+export function useDelegateKey() {
+  const [loading, setLoading] = useState<boolean>(false)
+  const [error, setError] = useState<Error | null>(null)
+  const { getPrivateKey, createMnemonicAccount } = useSign()
+
+  const createDelegate = useCallback(async (ownerAddress: Address) => {
+    setLoading(true)
+    setError(null)
+
+    try {
+      if (!ownerAddress) {
+        throw Logger.info('OwnerKeyNotFoundForDelegate')
+      }
+
+      // Step 0: Get users private key from keychain. TODO: need to pass ownerAddress to getPrivateKey
+      const privateKey = await getPrivateKey()
+
+      if (!privateKey) {
+        throw Logger.error('useDelegateKey: Something went wrong', error)
+      }
+
+      // Step 1: Generate delegate key
+      const delegateMnemonic = ethers.Wallet.createRandom().mnemonic?.phrase
+
+      if (!delegateMnemonic) {
+        throw Logger.error('useDelegateKey: Something went wrong', error)
+      }
+
+      const delegatedAccount = await createMnemonicAccount(delegateMnemonic)
+
+      if (!delegatedAccount) {
+        throw Logger.error('useDelegateKey: Something went wrong', error)
+      }
+
+      // Step 2: Generate message to sign
+      const time = Math.floor(Date.now() / 3600000).toString()
+      const messageToSign = delegatedAccount.address + time
+      const hashToSign = ethers.hashMessage(messageToSign)
+
+      // Step 3: Sign message
+      const signature = await signMessage({
+        signer: delegatedAccount,
+        message: hashToSign,
+      })
+
+      // Step 4: Send to backend
+      await createOnBackEnd(ownerAddress, delegatedAccount, signature)
+
+      // Step 5: Store delegate key in keychain
+      const encryptedDelegateKey = await DeviceCrypto.encrypt(asymmetricKey, delegatedAccount.privateKey, {
+        biometryTitle: 'Authenticate',
+        biometrySubTitle: 'Saving delegated key',
+        biometryDescription: 'Please authenticate yourself',
+      })
```

Comment:
we don't need this kind of protection for the delegate key. It is not going to sign transactions and can be exchanged at any point.

Reply:
No, I meant that we don't need biometry protection. Users should be able to remove the keys as in the old app.